### PR TITLE
BUGFIX: SETI-379 zuul-cloner tries to fetch a wrong revision

### PR DESCRIPTION
**Pull request by janpriessnitz**
Used str.replace instead of re.sub since regexes are not needed

### CODE REVIEW

**Comment by hrubi** [comment]
Please try to create a test for this, based on the problems mentioned in the ticket.

**Comment by janpriessnitz** [comment]
Should be OK now.

File: zuul/lib/cloner.py [hunk 1]
```diff
@@ -123,6 +122,21 @@ def fetchFromZuul(self, repo, project, ref):
                            project, ref)
             return False
 
+    def getOverrideRefs(self, project, indicated_branch, fallback_branch):
```
Comment (hrubi) [inline zuul/lib/cloner.py]: Unused `project`.

File: tests/test_cloner.py [hunk 2]
```diff
@@ -620,3 +620,23 @@ def test_post_and_master_checkout(self):
...
             repos[master_project].rev_parse('HEAD').hexsha,
             repos[master_project].rev_parse('master').hexsha)
+
+    def test_override_zuul_refs(self):
+        cloner = zuul.lib.cloner.Cloner(
+            git_base_url=self.upstream_root,
+            projects=['project1'],
+            workspace=self.workspace_root,
+            zuul_branch='7.2',
+            zuul_ref='refs/zuul/7.2/Z86a9b9068320423db7bb882627187.2b',
```
Comment (hrubi) [inline tests/test_cloner.py]: For the record, the original problem was with `zuul_ref='refs/zuul/7.2/Z86a9b9068320423db7bb882627187d2b'` where the `7d2` was matched regex-wise by the `zuul_branch=7.2`. You've dropped the regex usage, so here you are testing that the literal substitution happens only up to the last slash. And dot `.` is normally not a part of the random string. I'm just saying that this test is formally correct, but does not 100% match the practical use case.

**Comment by janpriessnitz** [comment]
Done. Now testing both the original bug and substitution up to the last slash.

**Comment by hrubi** [comment]
perfect.

**Pull request merged**